NN_npc.js

```diff
@@ -199,19 +407,96 @@ nerthus.npc.set_collision = function(npc)
         g.npccol[parseInt(npc.x) + 256 * parseInt(npc.y)] = true
 }
 
-nerthus.npc.load_npcs = function()
+nerthus.npc.set_collision_ni = function(npc)
+{
+    if(npc.collision)
+        Engine.map.col.set(parseInt(npc.x), parseInt(npc.y), 2)
+}
+
+
+nerthus.npc.reset_npcs = function ()
 {
-    var file_with_npc = nerthus.addon.fileUrl("/npcs/map_" + map.id + ".json")
+    $(".nerthus_npc").remove()
+}
+
+nerthus.npc.load_npcs = function ()
+{
+    const file_with_npc = nerthus.addon.fileUrl("/npcs/map_" + map.id + ".json")
     this.load_npcs_from_file(file_with_npc)
 }
 
-nerthus.npc.load_npcs_from_file = function(url)
+nerthus.npc.load_npcs_ni = function ()
 {
-    $.getJSON(url, function(npcs){npcs.forEach(nerthus.npc.deploy.bind(nerthus.npc))})
+    if (Engine.map.d.id === undefined)
+        setTimeout(this.load_npcs_ni.bind(this), 500)
+    else
+    {
+        nerthus.worldEdit.purgeNpcList()
+        let file_with_npc = nerthus.addon.fileUrl("/npcs/map_" + Engine.map.d.id + ".json")
+        this.load_npcs_from_file(file_with_npc)
+    }
 }
 
-nerthus.npc.start = function()
+nerthus.npc.load_npcs_from_file = function (url)
 {
-    nerthus.defer(this.load_npcs.bind(this))
+    $.getJSON(url, function (npcs)
+    {
+        if (npcs)
+            npcs.forEach(nerthus.npc.deploy.bind(nerthus.npc))
+    })
 }
 
+nerthus.npc.dialog.check = function (command)
+{
+    let match = command.match(/^talk.*id=(\d+)/)
+    if (match)
+    {
+        let id = match[1]
+        if (id >= 50000000)
+        {
+            if (this.list[id] !== undefined)
+            {
+                return id
+            }
+        }
+    }
+    return false
+}
+
+nerthus.npc.start = function ()
+{
+    nerthus.defer(function ()
+    {
+        nerthus.npc.load_npcs()
```

Comment:
czy load npc jest ttu potrzebne jak jest w callbacku na loadOnEveryMap?

Reply:
Z tego co widzę to nie, relikt z czasów kiedy był potrzebny bo nieco inaczej działała funkcja, usuwam.